app/services/scraper_service.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'selenium-webdriver'
+require 'nokogiri'
+
+class ScraperService < ApplicationService
+  attr_reader :keywords
+
+  def initialize(keywords)
+    @keywords = keywords
+  end
+
+  def call
```

Comment:
You can do something like this
```suggestion
class ScraperService < ApplicationService
  include Singleton
  
  def call(keyword)
```
and when you call `ScraperService` you use this instead `ScraperService.instance.call(keyword)`